footprints/main/views.py

```diff
@@ -244,6 +244,11 @@ def get_context_data(self, **kwargs):
         context['related'] = []
         context['editable'] = self.has_edit_permission(self.request.user)
         context['imprints'] = self.object.imprints()
+        context['state'] = {
+            'imprint': self.kwargs.get('imprint', None),
+            'copy': self.kwargs.get('copy', None),
+            'footprint': self.kwargs.get('footprint', None)
```

Comment:
The default for `.get()` is always `None`, so it's not necessary to specify that, but of course it doesn't hurt. https://docs.python.org/2/library/stdtypes.html#dict.get